web/components/editor/mention-suggestion.ts

```diff
@@ -1,21 +1,24 @@
 import type { MentionOptions } from '@tiptap/extension-mention'
 import { ReactRenderer } from '@tiptap/react'
-import { User } from 'common/user'
 import { searchInAny } from 'common/util/parse'
-import { orderBy } from 'lodash'
+import { memoize, orderBy } from 'lodash'
 import tippy from 'tippy.js'
+import { getUsers } from 'web/lib/firebase/users'
 import { MentionList } from './mention-list'
 
 type Suggestion = MentionOptions['suggestion']
 
+const users = memoize(getUsers)
```

Comment:
Seems like it would be better to use react query? The library kind of sucks if we can't cache the users and share that cache with other hooks, e.g. `useUserById`.

`useUserById` has the react query key ['users', userId]. The one you were using `useUsers` was just ['users']. So I think the cached results were not being shared between these hooks? Oh well.

I think you could replace this line with:
```
const queryClient = new QueryClient()
const users = () => queryClient.fetchQuery(['users'], getUsers)
```
Which would at least cache users for the useUsers hook.